### PR TITLE
Client side only mixins

### PR DESCRIPTION
**Pull request by Foghrye4**
Allow to start forge dedicated server.

### CODE REVIEW

File: src/main/java/cubicchunks/asm/mixin/core/client/MixinChunk_Cubes.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+/**
+ * Modifies vanilla code in Chunk to use Cubes. Client side only.
+ */
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+@Mixin(Chunk.class)
+@SideOnly(Side.CLIENT)
+public abstract class MixinChunk_Cubes implements IColumn {
+
+    // WARNING: WHEN YOU RENAME ANY OF THESE 3 FIELDS RENAME CORRESPONDING FIELDS IN MixinChunk_Column
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/asm/mixin/core/client/MixinChunk_Cubes.java]: Update this comment in all 3 classes that contain it

Comment (Foghrye4) [reply src/main/java/cubicchunks/asm/mixin/core/client/MixinChunk_Cubes.java]: @Barteks2x What do you mean? Point to full class path?

Comment (Niko-sk2x) [reply src/main/java/cubicchunks/asm/mixin/core/client/MixinChunk_Cubes.java]: The server version of this mixin and the one mentioned in this comment

File: src/main/java/cubicchunks/asm/mixin/core/client/MixinChunk_Cubes.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    // ==============================================
+
+    @Inject(method = "enqueueRelightChecks", at = @At(value = "HEAD"), cancellable = true)
+    private void enqueueRelightChecks_CubicChunks_NotSupported(CallbackInfo cbi) {
+        if (isColumn) {
+            // todo: enqueueRelightChecks
+            cbi.cancel();
+        }
+    }
+}
```
Comment (kashike) [inline src/main/java/cubicchunks/asm/mixin/core/client/MixinChunk_Cubes.java]: Missing newline at end of file.

Comment (Niko-sk2x) [reply src/main/java/cubicchunks/asm/mixin/core/client/MixinChunk_Cubes.java]: does anyone really care about that?

File: src/main/java/cubicchunks/asm/mixin/core/client/MixinChunk_Cubes.java (hunk 1 above)
Comment (Foghrye4) [reply src/main/java/cubicchunks/asm/mixin/core/client/MixinChunk_Cubes.java]: So I should change it to something like:

>    // WARNING: WHEN YOU RENAME ANY OF THESE 3 FIELDS RENAME CORRESPONDING FIELDS IN cubicchunks.asm.mixin.core.common.MixinChunk_Cubes and cubicchunks.asm.mixin.core.common.MixinChunk_Columns

for a client version and so on, yes?

Comment (Niko-sk2x) [reply src/main/java/cubicchunks/asm/mixin/core/client/MixinChunk_Cubes.java]: yes. All copies of them are actually the same thing at runtime.

File: src/main/java/cubicchunks/asm/mixin/core/client/MixinChunk_Cubes.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+/**
+ * Modifies vanilla code in Chunk to use Cubes. Client side only.
+ */
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+@Mixin(Chunk.class)
+@SideOnly(Side.CLIENT)
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/asm/mixin/core/client/MixinChunk_Cubes.java]: No need for these annotations, if it's specified only in the right mixin configurration.

File: src/main/java/cubicchunks/asm/mixin/core/client/MixinWorldProvider.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+@Mixin(WorldProvider.class)
+@SideOnly(Side.CLIENT)
```
Comment (Niko-sk2x) [inline src/main/java/cubicchunks/asm/mixin/core/client/MixinWorldProvider.java]: Same here

**Review by Niko-sk2x — CHANGES_REQUESTED** [review]

**Review by Niko-sk2x — APPROVED** [review]

**Pull request merged**